src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -124,44 +124,49 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
 
             _state.TcsQueryDataAvailable =
                 new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Maybe in the future `TcsQueryDataAvailable` could be changed to a subclass of `TaskCompletionSource` holding the relevant data to avoid the tuple allocation (similar to what was done in #7788), but for now I'm leaving this PR as-is.